src/main/java/org/mariadb/jdbc/MariaDbConnection.java

```diff
@@ -427,13 +427,10 @@ private boolean checkIfPreparable(String sql) {
         }
 
         String cleanSql = sql.toUpperCase().trim();
-        if (cleanSql.contains("SELECT")
+        return cleanSql.contains("SELECT")
                 || cleanSql.contains("UPDATE")
                 || cleanSql.contains("INSERT")
-                || cleanSql.contains("DELETE")) {
-            return true;
-        }
-        return false;
+                || cleanSql.contains("DELETE");
```

Comment:
that part doesn't concern directly your PR. I've added https://jira.mariadb.org/browse/CONJ-323 to handle this